fdbserver/workloads/ConsistencyCheck.actor.cpp

```diff
@@ -1933,6 +1938,89 @@ struct ConsistencyCheckWorkload : TestWorkload {
 		return true;
 	}
 
+	// Checks if the blob workers are "correct".
+	// Returns false if ANY of the following
+	// - any blob worker is on a diff DC than the blob manager/CC, or
+	// - any worker that should have a blob worker does not have one, or
+	// - any worker that should NOT have a blob worker does indeed have one
+	ACTOR Future<bool> checkBlobWorkers(Database cx,
+	                                    DatabaseConfiguration configuration,
+	                                    ConsistencyCheckWorkload* self) {
+		state vector<BlobWorkerInterface> blobWorkers = wait(getBlobWorkers(cx));
```

Comment:
nit: could optimize out this whole function if there are no blob workers.
In general we should try to make the performance overhead of all of our changes as minimal as possible, especially in the case when the blob feature isn't being used.

Reply:
Hmm, I wonder if it might be better to wait until we add the gate for blob-related work. Then we can actually assert that
```
blobGate <=> blobWorkers.size() > 0
```
Any thoughts? If that makes sense, I can add a TODO for that.

Also, is this a performance concern for when ConsistencyCheck becomes a role itself, as opposed to being restricted to testing?

Reply:
Sounds good.
So clarification though: the consistency check does already run in production at apple, for the data checking part.
It's just configured to run on test-class processes via the foundationdb.conf file as a test workload instead of a standalone role configured by the fdbcli. So that change won't affect this here.

Reply:
Ahh okay. I'll drop the TODO here so that we ASSERT based on the gate in the future, which I assume they will default to `false` to start with. That'll also serve the purpose of only doing the checks when there are indeed blob workers :)